pool/chainstate.go

```diff
@@ -102,7 +110,6 @@ func (cs *ChainState) handleChainUpdates(ctx context.Context) {
 			if err != nil {
 				log.Errorf("unable to create header from bytes: %v", err)
 				close(msg.Done)
-				cs.cfg.Cancel()
```

Comment:
While it is quite well explained in the test file, it's really not clear from just reading the source code here why some of the error cases are calling `Cancel()` and some of them are not. (For example - `PruneJobs()` error causes `Cancel()` to be called, and `AcceptedWorkID()` error does not.)

It might be worth explaining each case here as well, so a casual reader can understand without having to read both files.

Reply:
Thanks, I started combing through some more and found other cases that need better handing.